src/shared/splice-array.js

```diff
@@ -0,0 +1,20 @@
+const myPush = require('./push');
+
+function mySplicedArray(array, start, count) {
+  const { length } = array;
```

Comment:
Very interesting solution.
However - this is actually using "prototype" under the hood.
Take a look at this top answer:
https://stackoverflow.com/questions/42595907/how-does-destructuring-array-get-length-property

And also take a look at destructuring artcile:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

WOuld not count as serios mistake - but a minor one. For a clever solution.